File: hw/chicaiza/u3/hw36StrategyPersistence/hw35StrategyPersistence/hw Strategy/Java/CellphoneStore/src/main/java/utils/IdGenerator.java
package utils;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

public class IdGenerator {

    private static int counter = 0;

    
    public static String previewId() {
        LocalDate today = LocalDate.now();
        String datePart = today.format(DateTimeFormatter.ofPattern("yyyyMM"));
        String numberPart = String.format("%02d", counter);
        return datePart + numberPart;
    }

   
    public static String generateId() {
        if (counter > 99) {
            throw new IllegalStateException("Limit reached");
        }

        LocalDate today = LocalDate.now();
        String datePart = today.format(DateTimeFormatter.ofPattern("yyyyMM"));
        String numberPart = String.format("%02d", counter);

        counter++; 

        return datePart + numberPart;
    }
}

File: hw/chalan/u3/hw35StrategyPersistence/hw Strategy/Python/cellphone_store/utils/id_generator.py
from datetime import datetime

class IdGenerator:
    _counter = 0

    @staticmethod
    def preview_id():
        now = datetime.now()
        date_part = now.strftime("%Y%m")
        number_part = f"{IdGenerator._counter:02d}"
        return date_part + number_part

    @staticmethod
    def generate_id():
        if IdGenerator._counter > 99:
            raise Exception("Limit reached")

        now = datetime.now()
        date_part = now.strftime("%Y%m")
        number_part = f"{IdGenerator._counter:02d}"

        IdGenerator._counter += 1
        return date_part + number_part

